src/main/scala/cromwell/engine/package.scala

```diff
@@ -34,4 +34,9 @@ package object engine {
     override def toString: String = "Succeeded"
     override val isTerminal = true
   }
+
+  trait WorkflowDescriptor {
+    def name: String
+    def id: WorkflowId
+  }
```

Comment:
Before throwing the kitchen sink in there I'd like to see a definition of what this construct is supposed to embody and then when adding things there should be a decent argument for it. I have followups for most answers to that, but not all :)